ginivision/src/androidTest/java/net/gini/android/vision/review/ReviewScreenTest.java

```diff
@@ -378,4 +380,40 @@ public void should_finishIfEnabledByClient_whenReceivingActivityResult_withResul
 
         verify(reviewActivitySpy).finish();
     }
+
+    @Test
+    public void
+    should_returnDocuments_withSameUUID_inAnalyzeDocument_andProceedToAnalysis()
+            throws InterruptedException {
+        final ReviewActivityTestStub activity = startReviewActivity(TEST_JPEG, 90);
+
+        final AtomicReference<Document> documentToAnalyze = new AtomicReference<>();
+        final AtomicReference<Document> documentToProceedWith = new AtomicReference<>();
+
+        activity.setListenerHook(new ReviewActivityTestStub.ListenerHook() {
+            @Override
+            public void onShouldAnalyzeDocument(@NonNull Document document) {
+                documentToAnalyze.set(document);
+            }
+
+            @Override
+            public void onProceedToAnalysisScreen(@NonNull final Document document) {
+                documentToProceedWith.set(document);
+            }
+        });
+
+        // Modify the document
+        Espresso.onView(ViewMatchers.withId(R.id.gv_button_rotate))
+                .perform(ViewActions.click());
+
+        // Click next
+        Espresso.onView(ViewMatchers.withId(R.id.gv_button_next))
+                .perform(ViewActions.click());
+
+        // Allow the activity to run a little for listeners to be invoked
+        Thread.sleep(PAUSE_DURATION);
```

Comment:
Haha, a confessed sin is half a sin? I love it.
BTW, exposing internal bug tracking URLs to public repositories is probably not super nice.